### PR TITLE
MODINREACH-281 - Store setting to indicate whether to look up pickup locations for INN-Reach item hold requests based on transaction pickupLocation

### PR DESCRIPTION
**Pull request by s1mpleGit**
[MODINREACH-281](https://issues.folio.org/browse/MODINREACH-281) - Store setting to indicate whether to look up pickup locations for INN-Reach item hold requests based on transaction pickupLocation

## Purpose
Store a per central server setting that indicates whether mod-inn-reach should first attempt to look up a pickup location for the FOLIO request it creates for an Item Hold transaction using the pickup location code from the Item Hold transaction (similar to what is done for patron hold requests) or simply rely on existing logic (use the default pickup location of the institutional patron for which the request is created).

## Approach
Central server schema changed
Central server settings schema added
Liquibase scripts changed
Tests added

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "Central server configuration",
+  "type": "object",
+  "properties": {
+    "centralServerId": {
+      "description": "Central server id",
+      "type": "string",
+      "format": "UUID"
+    },
+    "checkPickupLocation": {
+      "description": "Check for pickup location",
```
Comment (oliinyko) [inline src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json]: Just use the definition from story.
indicate whether to look up pickup locations for INN-Reach item hold requests based on transaction pickupLocation

File: src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      "type": "boolean"
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "centralServerId",
+    "checkPickupLocation"
```
Comment (oliinyko) [inline src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json]: I don't think it should be required, the default value should be false

File: src/main/resources/swagger.api/schemas/centralServerDTO.json [hunk 3]
```diff
@@ -57,6 +57,10 @@
       "description": "Local server secret",
       "type": "string"
     },
+    "checkPickupLocation": {
+      "description": "Check which pickup location to choose",
+      "type": "boolean"
```
Comment (oliinyko) [inline src/main/resources/swagger.api/schemas/centralServerDTO.json]: default: false

File: src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json (hunk 1 above)
Comment (s1mpleGit) [reply src/main/resources/swagger.api/schemas/centralServerSettingsDTO.json]: replaced

File: src/main/resources/swagger.api/schemas/centralServerDTO.json (hunk 3 above)
Comment (s1mpleGit) [reply src/main/resources/swagger.api/schemas/centralServerDTO.json]: added

**Review by oliinyko — APPROVED** [review]

**Pull request merged**